src/Http/Http.Results/src/ProblemHttpResult.cs

```diff
@@ -57,6 +57,7 @@ public Task ExecuteAsync(HttpContext httpContext)
                 httpContext,
                 logger,
                 value: ProblemDetails,
+                valueType: ProblemDetails.GetType(),
```

Comment:
Instead of adding a new overload, would it be simpler just to change the call to the existing overload to be `HttpResultsHelper.WriteResultAsJsonAsync<object>(`?

Reply:
Thanks. I've completely missed this option.